centreon/src/Core/HostTemplate/Application/UseCase/AddHostTemplate/AddHostTemplateValidation.php

```diff
@@ -0,0 +1,215 @@
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\HostTemplate\Application\UseCase\AddHostTemplate;
+
+use Centreon\Domain\Contact\Interfaces\ContactInterface;
+use Centreon\Domain\Log\LoggerTrait;
+use Core\Command\Application\Repository\ReadCommandRepositoryInterface;
+use Core\Common\Domain\CommandType;
+use Core\HostCategory\Application\Repository\ReadHostCategoryRepositoryInterface;
+use Core\HostSeverity\Application\Repository\ReadHostSeverityRepositoryInterface;
+use Core\HostTemplate\Application\Exception\HostTemplateException;
+use Core\HostTemplate\Application\Repository\ReadHostTemplateRepositoryInterface;
+use Core\HostTemplate\Domain\Model\HostTemplate;
+use Core\Security\AccessGroup\Application\Repository\ReadAccessGroupRepositoryInterface;
+use Core\TimePeriod\Application\Repository\ReadTimePeriodRepositoryInterface;
+use Core\Timezone\Application\Repository\ReadTimezoneRepositoryInterface;
+use Core\ViewImg\Application\Repository\ReadViewImgRepositoryInterface;
+
+class AddHostTemplateValidation
+{
+    use LoggerTrait;
+
+    public function __construct(
+        private readonly ReadHostTemplateRepositoryInterface $readHostTemplateRepository,
+        private readonly ReadViewImgRepositoryInterface $readViewImgRepository,
+        private readonly ReadTimePeriodRepositoryInterface $readTimePeriodRepository,
+        private readonly ReadHostSeverityRepositoryInterface $readHostSeverityRepository,
+        private readonly ReadTimezoneRepositoryInterface $readTimezoneRepository,
+        private readonly ReadCommandRepositoryInterface $readCommandRepository,
+        private readonly ReadHostCategoryRepositoryInterface $readHostCategoryRepository,
+        private readonly ReadAccessGroupRepositoryInterface $readAccessGroupRepository,
+        private readonly ContactInterface $user
+    ) {
+    }
+
+    /**
+     * Assert name is not already used.
+     *
+     * @param string $name
+     *
+     * @throws HostTemplateException
+     */
+    public function assertIsValidName(string $name): void
+    {
+        $formattedName = HostTemplate::formatName($name);
+        if ($this->readHostTemplateRepository->existsByName($formattedName)) {
+            $this->error('Host template name already exists', ['name' => $name, 'formattedName' => $formattedName]);
```

Comment:
Just for knowledge about [compact](https://www.php.net/manual/en/function.compact.php) 😉 
💯 equivalent in this case.
```suggestion
            $this->error('Host template name already exists', compact('name', 'formattedName'));
```